Ginger/GingerCoreNET/LiteDBFolder/LiteDbConnector.cs

```diff
@@ -31,10 +32,28 @@ namespace Amdocs.Ginger.CoreNET.LiteDBFolder
     //Need to make sure that LiteDb handles it internally
     public class LiteDbConnector
     {
-        public string ConnectionString { get; set; }
+        public ConnectionString ConnectionString { get; set; }
         public LiteDbConnector(string filePath)
         {
-            this.ConnectionString = $"filename={filePath}; connection=Shared; upgrade=true";
+            ConnectionString = new()
+            {
+                Filename = filePath,
+                Connection = ConnectionType.Shared
+            };
+            TryUpgradeDataFile();
+        }
+
+        private bool TryUpgradeDataFile()
+        {
+            try
+            {
+                string dbFilePath = ConnectionString.Filename;
+                return LiteEngine.Upgrade(dbFilePath);
+            }
+            catch(Exception)
+            {
+                return false;
+            }
```

Comment:
it is possible that the calling class not provide file while creating the object, and set it later. In that case this method would throw an exception and it will get logged unnecessarily.